### PR TITLE
feat: ( os ) Implement FileStateSync and test in test.ts

### PR DESCRIPTION
**Pull request by Foveluy**

### CODE REVIEW

File: os.go [hunk 1]
```diff
@@ -193,3 +195,22 @@ func WriteFileSync(filename string, data []byte, perm uint32) []byte {
 	check(err)
 	return out
 }
+
+func StatSync(filename string) []byte {
+	info, err := os.Stat(filename)
```
Comment (qti3e) [inline os.go]: Use `afero.Stat`

Comment (Foveluy) [reply os.go]: @qti3e Sure, no problem.

Thanks for reviewing.

**Comment by Foveluy** [comment]
Done and also add `tests` in `tests.ts`.

File: deno.d.ts [hunk 2]
```diff
@@ -1,5 +1,6 @@
 // Copyright 2018 Ryan Dahl <ry@tinyclouds.org>
 // All rights reserved. MIT License.
+import { FileInfo } from './types'
```
Comment (yorkie) [inline deno.d.ts]: Miss a ;

Comment (Foveluy) [reply deno.d.ts]: @yorkie great...! Thanks

**Comment by Kragwun** [comment]
Am replying

File: types.ts [hunk 3]
```diff
@@ -8,3 +8,11 @@ export interface ModuleInfo {
   sourceCode?: string;
   outputCode?: string;
 }
+
+export interface FileInfo {
+  filename?: string;
+  isDir?: boolean;
+  size?: number | Long;
+  modTime?: number | Long;
```
Comment (ry) [inline types.ts]: Why "Long" ?

File: os.ts [hunk 4]
```diff
@@ -51,6 +51,20 @@ export function readFileSync(filename: string): Uint8Array {
   return res.readFileSyncData;
 }
 
+export function fileStatSync(filename: string): FileInfo {
```
Comment (ry) [inline os.ts]: Rename to `statSync`

File: types.ts (hunk 3 above)
Comment (Foveluy) [reply types.ts]: @ry I have fixed conflicts files, could you please review again ?

Thanks.

**Comment by ry** [comment]
Sorry for ignoring this - we have been basically working on a completely new implementation. This is now not mergeable. We will revisit the issue in a few days once the runtime on rust is working.
Thanks for the effort @Foveluy

**Pull request closed without merging**